File: ts-sdk/src/sdk/models/operations/updatewallet.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class UpdateWalletPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}

export class UpdateWalletRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;
}

export class UpdateWalletRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateWalletPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: UpdateWalletRequestBody;
}

export class UpdateWalletResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  walletsErrorResponse?: shared.WalletsErrorResponse;
}

File: python-sdk/src/formanceapi/models/operations/updatewallet.py
import dataclasses
from ..shared import walletserrorresponse as shared_walletserrorresponse
from dataclasses_json import dataclass_json
from formanceapi import utils
from typing import Any, Optional


@dataclasses.dataclass
class UpdateWalletPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class UpdateWalletRequestBody:
    metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    

@dataclasses.dataclass
class UpdateWalletRequest:
    path_params: UpdateWalletPathParams = dataclasses.field()
    request: Optional[UpdateWalletRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class UpdateWalletResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    wallets_error_response: Optional[shared_walletserrorresponse.WalletsErrorResponse] = dataclasses.field(default=None)
    